### ISSUE TITLE
After rpm upgrade on centos 6, sync_gateway binary is missing

### ISSUE THREAD

**Issue by ajres** [couchbase/sync_gateway#1160]
On Centos six, tried to upgrade Sync Gateway from 1.1.0 enterprise to 1.1.1 enterprise using rpm package upgrade.

```
# rpm -U couchbase-sync-gateway-enterprise_1.1.1-10_x86_64.rpm
```

Expected result:

Sync Gateway would be upgraded

Actual result:

The Sync Gateway binary file is missing:

```
/opt/couchbase-sync-gateway/bin/sync_gateway
```

Upgrade works if old version is first removed and the new version installed

```
# rpm -e couchbase-sync-gateway
# rpm -i couchbase-sync-gateway-enterprise_1.1.1-10_x86_64.rpm
```

**Comment by ajres** [comment]
The issue is that the pre/post uninstall scripts for version 1.1.0 are called after the pre/post install scripts for  1.1.1.

This results int he sync_gateway binary being removed as part of the 1.1.0 uninstall

**Comment by ajres** [comment]
%pre/post uninstall scriptlets are called with a parameter to indicate whether they are being called as part of a removal or an upgrade, this is now interrogated and no delete is executed during an upgrade.

Merged back to a branch from master and PR #1249 raised.

Closing out this ticket.
